deepchem/feat/protein_featurizers/protein_sequence_featurizer.py

```diff
@@ -0,0 +1,138 @@
+import os
+import numpy as np
+import tempfile
+from amino_acids import AminoAcid
+from deepchem.utils.protein_sequence_data_utils import read_fasta
+from deepchem.utils.protein_sequence_data_utils import read_hhm
+from deepchem.utils.protein_sequence_data_utils import run_hhblits_local
+from deepchem.utils.protein_sequence_feature_utils import read_a3m_as_mat
+from deepchem.utils.protein_sequence_feature_utils import sequence_one_hot_encoding
+from deepchem.utils.protein_sequence_feature_utils import sequence_deletion_probability
+from deepchem.utils.protein_sequence_feature_utils import sequence_weights
+from deepchem.utils.protein_sequence_feature_utils import sequence_profile
+from deepchem.utils.protein_sequence_feature_utils import sequence_profile_no_gap
+from deepchem.utils.protein_sequence_feature_utils import sequence_profile_with_prior
+from deepchem.utils.protein_sequence_feature_utils import sequence_identity
+from deepchem.utils.protein_sequence_feature_utils import sequence_static_prop
+from deepchem.utils.protein_sequence_feature_utils import sequence_gap_matrix
+from deepchem.utils.protein_sequence_feature_utils import profile_combinatorial
+from deepchem.utils.protein_sequence_feature_utils import mutual_information
+from deepchem.utils.protein_sequence_feature_utils import mean_contact_potential
+
+from deepchem.feat.base_classes import Featurizer
+
+
+
+class ProteinSequenceFeaturizer(Featurizer):
+  """Abstract class for calculating a set of features for a
+  protein sequence (used for structure prediction).
+
+  The defining feature of a `ProteinSequenceFeaturizer` is that it
+  reads protein sequences (and pre-saved multiple sequence alignments)
```

Comment:
It might be useful to add a couple references here for what a multiple sequence alignment is and why it's interesting